src/examples/libpmemset/basic/basic.c

```diff
@@ -0,0 +1,169 @@
+	struct pmemset_part_descriptor desc[NUMBER_OF_PARTS];
+	struct pmemset_part_map *part_maps[NUMBER_OF_PARTS];
+
+	/* Parse and validate input arguments, file path is obligatory. */
+	if (argc != 2) {
+		fprintf(stderr, "usage: %s file\n", argv[0]);
+		exit(1);
+	}
+
+	char *file = argv[1];
+	int ret;
+
+	/*
+	 * This function creates pmemset source using a file path.
+	 *
+	 * The libpmemset library also allows creating source
+	 * using file descriptor, pmem2 source, or temporary file.
+	 * For more details, see pmemset_source_from_fd(3),
+	 * pmemset_source_from_pmem2(3) and pmemset_source_from_temporary(3)
+	 * man pages.
+	 */
+	ret = pmemset_source_from_file(&src, file);
+	if (ret) {
+		pmemset_perror("pmemset_source_from_file");
+		exit(1);
+	}
+
+	/*
+	 * This function initializes the config of the entire set.
+	 * The only required parameter in the set configuration is
+	 * granularity. But more attributes can be specified eg.
+	 * memory reservation, events, acceptable part states,
+	 * part coalescing.
+	 *
+	 * For more details about the granularity concept, especially other
+	 * types like PMEM2_GRANULARITY_CACHE_LINE, PMEM2_GRANULARITY_BYTE see
+	 * libpmem2(7) man page.
+	 */
+	ret = pmemset_config_new(&cfg);
+	if (ret) {
+		pmemset_perror("pmemset_config_new");
+		goto exit_src_del;
+	}
+
+	/* Set required store granularity in the config. */
+	enum pmem2_granularity graunlarity = PMEM2_GRANULARITY_PAGE;
+	ret = pmemset_config_set_required_store_granularity(cfg, graunlarity);
+	if (ret) {
+		pmemset_perror("pmemset_config_set_required_store_granularity");
+		goto exit_cfg_del;
+	}
+
+	/* Create a new set object using previously defined config. */
+	ret = pmemset_new(&set, cfg);
+	if (ret) {
+		pmemset_perror("pmemset_new");
+		goto exit_cfg_del;
+	}
+
+	/*
+	 * This function createss new map configuration.
+	 *
+	 * Map configuration is not required to create a new mapping,
+	 * however, can extend the functionality by defining the mapping length
+	 * and offset in the file. By default entire size would be used.
+	 */
+	ret = pmemset_map_config_new(&map_cfg);
+	if (ret) {
+		pmemset_perror("pmemset_map_config_new");
+		goto exit_set_del;
+	}
+
+	/* Configure the size of the new mapped part. */
+	pmemset_map_config_set_length(map_cfg, PART_SIZE);
+
+	/*
+	 * Map a few parts based on the prepared configuration.
+	 *
+	 * The last parameter to the pmemset_map function is also optional.
+	 * It represents part descriptor - structure describing the
+	 * created mapping. There is also another way to read
+	 * information about this structure, using the function
+	 * pmemset_descriptor_part_map(3) directly on created part_map,
+	 * as in the example below.
+	 */
+	for (int i = 0; i < NUMBER_OF_PARTS; i++) {
+		ret = pmemset_map(set, src, map_cfg, NULL);
+		if (ret) {
+			pmemset_perror("pmemset_map");
+			goto exit;
+		}
+	}
+
+	/* Get first part map from the set. */
+	pmemset_first_part_map(set, &part_maps[0]);
```

Comment:
I think I'd prefer we did this in a for-loop style:

`for (part = first_part_map; part != NULL; part = next_part_map) {}`

(but now that I wrote it I'm not sure - what does everyone else think?)